### ISSUE TITLE
Getting null for newly added elements in Cache.

### ISSUE THREAD

**Issue by kuldeep9922** [hazelcast/hazelcast#7634]
Here's the map configuration:

```
<hz:map name="cacheName"
	                backup-count="1"
	                max-size="10000"
	                eviction-percentage="0"
	                read-backup-data="true"
	                eviction-policy="LRU"
	                merge-policy="com.hazelcast.map.merge.LatestUpdateMapMergePolicy"
	                time-to-live-seconds="0"
	                >
</hz:map>
```

After reaching max-size of 10000, it starts return null object for newly added values in cache. There seems to be some problem with eviction in hazel-cast. How can I  fetch newly added values?

**Comment by ahmetmircik** [comment]
Hi @kuldeep9922,

This is a known issue https://github.com/hazelcast/hazelcast/issues/4334 and scheduled to fix in 3.7. As a workaround you can increase `eviction-percentage` to a higher number (for example 25) and listen local entry evicted events if the evicted key is a fresh key, you can try to put it again into `IMap`. (Btw, i didn't try this workaround, only i thought it may work)

**Comment by kuldeep9922** [comment]
Hi @ahmetmircik,

Thanks for the reply. We already tried this workaround and it's not working. Is there any other way possible as of now?

**Comment by ahmetmircik** [comment]
Hi, it will be fixed in 3.7. As an other alternative, if you can use jcache instead of imap, this problem should not be happen.

**Comment by d123456temp** [comment]
This is definitely NOT fixed for the LRU strategy. Newly added items are still being evicted straight away before really old entries when Map is working at almost-full size (so it reached max-elements-count).
This is the first bad experience I had with HAzelcast. It would be a shame as this is pretty much very important, basic functionality that should work without any issues.
